mongodbatlas/fw_resource_mongodbatlas_database_user.go

```diff
@@ -0,0 +1,529 @@
+						"role_name": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"labels": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+						"value": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"scopes": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Optional: true,
+						},
+						"type": schema.StringAttribute{
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func NewDatabaseUserRS() resource.Resource {
+	return &DatabaseUserRS{}
+}
+
+func (r *DatabaseUserRS) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, databaseUserResourceName)
+}
+
+func (r *DatabaseUserRS) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	client, err := ConfigureClientInResource(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+	r.client = client
+}
+
+func (r *DatabaseUserRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var databaseUserPlan *tfDatabaseUserModel
+
+	diags := req.Plan.Get(ctx, &databaseUserPlan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	dbUserReq, d := newMongoDBDatabaseUser(ctx, databaseUserPlan)
+	resp.Diagnostics.Append(d...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.client.Atlas
+	dbUser, _, err := conn.DatabaseUsers.Create(ctx, databaseUserPlan.ProjectID.ValueString(), dbUserReq)
+	if err != nil {
+		resp.Diagnostics.AddError("error during database user creation", err.Error())
+		return
+	}
+
+	dbUserModel, diagnostic := newTFDatabaseUserModel(ctx, databaseUserPlan, dbUser)
+	resp.Diagnostics.Append(diagnostic...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &dbUserModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *DatabaseUserRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var databaseUserState *tfDatabaseUserModel
+	var err error
+	resp.Diagnostics.Append(req.State.Get(ctx, &databaseUserState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	username := databaseUserState.Username.ValueString()
+	projectID := databaseUserState.ProjectID.ValueString()
+	authDatabaseName := databaseUserState.AuthDatabaseName.ValueString()
+
+	// Use the ID only with the IMPORT operation
+	if databaseUserState.ID.ValueString() != "" && (username == "" || projectID == "" || authDatabaseName == "") {
```

Comment:
Could you explain how this works with import only? How are we sure that ID will be used only with IMPORT operation? I believe we add `Id` in the same format during CREATE/update as well

Reply:
sure, if the id field is populated and one of the required fields is empty, we know is from the import as the user has to provide the required fields during the creation.